hack/release-manifest.py

```diff
@@ -0,0 +1,103 @@
+import os
+import yaml
+import sys
+
+
+OPERATOR_VERSION = os.getenv('OPERATOR_VERSION')
+GIT_COMMIT_ID = os.getenv('GIT_COMMIT_ID')
+release_yaml = "./out/releases/" + OPERATOR_VERSION + "/release.yaml"
+
+def release_manifest(): 
```

Comment:
IMHO, the script can be simplified significantly, if we use `yaml` library more.

https://stackoverflow.com/questions/53150743/add-multiple-docs-in-yaml-file-pyyaml

and for example `glob` package:

https://pymotw.com/2/glob/

we need to perform the following:
* iterate over all `*.yaml` files in the passed manifest folder
* load `deploy/service_account.yaml` and add it to the list
* if the filename does not end with `clusterserviceversion.yaml`, read it using `yaml` library, store in the list
* for CSV file, read it using `yaml` library and extract from there (1) `spec.install.spec.deployments in standalone deployments, and (2) `spec.install.permissions.rules` into standalone cluster role
* create cluster role binding using `deploy/role_binding.yaml`

We should not use `deploy/operator.yaml` - the real deployment with real image ref is already part of CSV, we should extract it from there.

Reply:
I tried the yaml.dump_all function and it gives a stream which can be hard to read for a user. It gives something like
```
      example-servicebinding\"\n    },\n    \"spec\": {\n      \"application\": {\n\
      \        \"group\": \"apps\",\n        \"name\": \"nodejs-rest-http-crud\",\n\
      \        \"resource\": \"deployments\",\n        \"version\": \"v1\"\n     \
      \ },\n      \"services\": [\n        {\n          \"group\": \"postgresql.example.dev\"\
      ,\n          \"kind\": \"Database\",\n          \"name\": \"pg-instance\",\n\
      \          \"version\": \"v1alpha1\"\n        }\n      ]\n    }\n  }\n]"
      ```

Reply:
This would work but the user who might take a look at the release.yaml at a glance, this might be not so readable

Reply:
> I tried the yaml.dump_all function and it gives a stream which can be hard to read for a user. It gives something like

What you posted is not yaml, it looks like json.

I have just tried to follow stackoverflow link shared above and I was able to get proper yaml files. It would be interesting to see the code snippet that generates the snippet you posted.

Reply:
This conversation is marked as resolved, but unfortunately the script still need to evolve to address the previous comments, so that readability and maintainability can reach acceptable level:

* we should iterate over all found yaml files, not only fixed one.
* we should not copy files line be line
* ClusterRole should be constructed from CSV, no need to keep it in `deploy` folder.
* there is no need to use temp file `.data.yaml`

Reply:
PTAL https://github.com/redhat-developer/service-binding-operator/pull/806/commits/10c7cabca396b5e1f0435bb5f25f5115db807cda